tensordict/tensordict.py

```diff
@@ -153,12 +154,13 @@ def __iter__(self):
 
     def _iter_helper(self, tensordict, prefix=None):
         items_iter = self._items(tensordict)
-
+        self.loop_set.add(id(tensordict))
```

Comment:
`_TensorDictKeysView` is already a compute bottleneck in many cases and I'm afraid that this set manipulation adds a consequent overhead to the whole thing.
We could make make a list instead and iterate over the reversed list:
```python
self.loop_list = []
...
self.loop_list.append(id(tensordict))
...
  and id(value) not in reversed(self.loop_set)
...
self.loop_list = self.loop_list[:-1]
```
(not sure I get the full picture or that would be significantly faster).

If you have ideas about how to make this faster or some tests that show that for deeply nested tensordicts we don't have a significant overhead with the set I'd be supportive of it.

Reply:
It's not just one way traversal. You should keep all states you went through. For ex: 
<img width="200" alt="image" src="https://user-images.githubusercontent.com/11034692/210366863-5224f479-fb22-443e-ab79-014e23dbf2a2.png">
Each node is tensordict and arrows are drawn if one TD contains another TD. If we go through 1->2->3->4 and then return add remove from loop_list all this nodes we will go to node 5 and then again to node 4. But if we keep it in our set there will not be this problem.  Also because it's not one way traversal in terms of complexity it's cheaper to use set with (O(1) in amortized complexity).

Reply:
Makes sense! thanks